src/home/CallList.tsx

```diff
@@ -55,8 +58,16 @@ interface CallTileProps {
   client: MatrixClient;
 }
 
-const CallTile: FC<CallTileProps> = ({ name, avatarUrl, room }) => {
+const CallTile: FC<CallTileProps> = ({ name, avatarUrl, room, client }) => {
   const roomEncryptionSystem = useRoomEncryptionSystem(room.roomId);
+  const onRemove = useCallback(
+    (e: MouseEvent) => {
+      e.stopPropagation();
+      e.preventDefault();
+      void client.leave(room.roomId);
```

Comment:
This works since we listen to room member changes and update the list reactivity.
This also means, that on click we send the leave but the room is still rendered in the list.
The user can trigger the leave multiple times by pressing the button quickly?
What would be really pretty is disabling the remove button right on click and enabling it if an error occurred:
we would need to introduce a `[leaving, setLeaving] = useState(false)`
```suggestion
      setLeaving(true);
      client.leave(room.roomId).catch(()=>setLeaving(false));
```
And use leaving to change the IconButtons Enabled state. (If you want to go really fancy you can even change the transparency of the `CallTile`.
But this might want to get feedback from design @amshakal.
Disabling the button is almost non controversial imo.

Reply:
Good call. I've went a little further than this because the link itself has the same problem and is still clickable while leaving. Unfortunately, the `Link` component doesn't appear to support disabling so I've conditionally replaced it with a span. For the disabled look of the link, I simply reused the selection style (80% opacity).